mobile/library/cc/engine_builder.h

```diff
@@ -104,6 +104,7 @@ class EngineBuilder {
 
   EngineSharedPtr build();
 
+  bool generateBootstrapAndCompare(std::string yaml) const;
```

Comment:
nit: Can you add a comment for this method? Also instead of std::string we should probably pass in absl::string_view. Ditto for generateBootstrapAndCompareForTests.

Reply:
It looks like all the other functions in this file use `std::string` instead of `absl::string_view`. Is this one special or should this have been done for all APIs?

Reply:
It depends a bit. We want to avoid copies where possible. For a setter where we have a member which is an std::string, a copy is inevitable. But the question is where it happens. If we take an std::string by value in the method then we can do `member_ = std::move(arg)`. This avoids a copy in the method. If the caller has a string, the caller can do `set_foo(std:move(data))` to avoid a copy (assuming they have the data in a variable that can be moved). So for setters, std::string is the best choice.

But for an argument where we don't need to copy the data, passing `absl::string_view` (or `const std::string&`, though if the caller has a char* that might require a copy) is better. In this case, I think we don't need to copy `yaml` so passing a copy of it is not necessary. 

What do you think?

Reply:
Yeah avoiding copies is a good reason to do this, I'm still getting used to C++'s model for implicit copying. Thanks for explaining.

Reply:
You're welcome! C++ is just astonishingly complex and with new versions of C++ the best practices keep changing. So on good days I feel like I almost maybe have a handle on it. Sorta :)